pages/api/[userName].js

```diff
@@ -0,0 +1,13 @@
+import profileData from '../../server/profile-data.json';
+
+export default function handler(req, res) {
+  const { userName } = req.query;
+  if (userName) {
+    const user = profileData.find((u) => u.user_name === userName);
+    if (user) {
+      res.status(200).json([user]);
+    } else {
+      res.status(404).send('Not found.');
+    }
+  }
+}
```

Comment:
This file name will cause issues for you. Because it will catch everything `/api/blabla`. Please change it to `api/users.js` and use the query param instead. Luckily, in this case, you only need to rename the file.